funfact/lang/interpreter/_vectorize.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import copy
+from typing import Optional
+from funfact.util.iterable import as_tuple
+from funfact.lang._ast import _AST, _ASNode, Primitives as P
+from funfact.lang._terminal import AbstractIndex, AbstractTensor, LiteralValue
+from ._base import _deep_apply, TranscribeInterpreter
+
+
+class Vectorizer():
```

Comment:
This class can subclass TranscribeInterpreter, right? Then we don't need to redefine `__call__` again.